src/server/main_service.cc

```diff
@@ -1038,6 +1038,79 @@ void Service::Function(CmdArgList args, ConnectionContext* cntx) {
   return (*cntx)->SendError(err, kSyntaxErrType);
 }
 
+vector<std::string_view> UnionResultVec(const vector<vector<string_view>>& result_vec) {
+  absl::flat_hash_set<std::string_view> uniques;
+
+  for (const auto& val : result_vec) {
+    for (const string_view& s : val) {
+      uniques.emplace(s);
+    }
+  }
+
+  vector<std::string_view> result;
+  result.reserve(uniques.size());
+  copy(uniques.begin(), uniques.end(), back_inserter(result));
+  return result;
+}
+
+void Service::PubsubChannels(string_view pattern, ConnectionContext* cntx) {
+  vector<vector<string_view>> result_set(shard_set->size());
+
+  shard_set->RunBriefInParallel([&](EngineShard* shard) {
```

Comment:
You do not need `UnionResultVec` since channels are shared among all shards, hence the set is always disjoint. Therefore you just need to add their cardinality. Similar to the comment below, you do not need `ListChannels`, you need `ChannelsCount` and then sum up all the counts across all shards.

Reply:
Pubsub channels returns an array of channels, not count.